ci/make_container_images.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+# This script is intended to be used by Cirrus-CI, from the VM
+# built by the 'image_builder' makefile target in this repo.  Use
+# of this script in any other context/environment is unlikely to
+# function as intended.
+
+set -e
+
+SCRIPT_FILEPATH=$(realpath "$0")
+SCRIPT_DIRPATH=$(dirname "$SCRIPT_FILEPATH")
+REPO_DIRPATH=$(realpath "$SCRIPT_DIRPATH/../")
+
+# shellcheck source=./lib.sh
+source "$REPO_DIRPATH/lib.sh"
+
+# When running under Cirrus-CI on a branch, also tag/push "latest" images
+PUSH_LATEST=0
+
+# shellcheck disable=SC2154
+if [[ -z "$CI" ]] || [[ "$CI" != "true" ]] || [[ "$CIRRUS_CI" != "$CI" ]]; then
+    die "Unexpected \$CI=$CI and/or \$CIRRUS_CI=$CIRRUS_CI"
+elif [[ -z "$IMG_SFX" ]]; then
+    die "Required non-empty values for \$IMG_SFX=$IMG_SFX"
+elif [[ -z "$TARGET_NAME" ]]; then
+    die "Required makefile target \$TARGET_NAME to be non-empty"
+elif [[ -z "$DEST_FQIN" ]]; then
+    die "Required destination FQIN \$DEST_FQIN to be non-empty"
+# avoid exposing secret contents
+elif [[ "${#LOGIN_CMD}" -le 0 ]] || grep -Eq "ENCRYPTED" <<<"$LOGIN_CMD"; then
+    die "Required \$LOGIN_CMD to be decrypted and longer than ${#LOGIN_CMD} characters"
+fi
+
+set -eo pipefail
+make "$TARGET_NAME" IMG_SFX=$IMG_SFX
+
+set +x
+# Cirrus sets CIRRUS_BRANCH="pull/#" for pull-requests, branch-name for branches
+# shellcheck disable=SC2154
+if [[ -n "$CIRRUS_BRANCH" ]] && [[ -z "$CIRRUS_PR" ]] && [[ -z "$CIRRUS_TAG" ]]; then
+    PUSH_LATEST=1
+fi
+
+trap "podman logout --all" EXIT INT CONT
+${LOGIN_CMD}
```

Comment:
This is not something to fix now, but definitely in future such work: I am very uncomfortable with the "podman logout". Up until this line I had no idea what \$LOGIN_CMD was, I just shrugged and let it pass. Now I realize it must be a "podman login" command ... and it just feels wrong to have a "podman logout" without a matching login. In future work I think it would be more maintainable to do:
```bash
    podman login --user $DECRYPTED_PODMAN_REGISTRY_USER \
        --password-stdin \
        $REMOTE_REGISTRY <<<"$DECRYPTED_PODMAN_REGISTRY_PASSWORD"
```
(with better variable names of course; those are just for clarity here)

Reply:
FWIW: `podman logout` always works, even if you're not logged in.  I put everything in one var. out of pure laziness :smile: Your way is certainly more readable though, I get your point.